tools/adventure-pack/src/app/__tests__/__snapshots__/equip-test.ts.snap

```diff
@@ -1439,6 +1439,26 @@ function* traverseInorder(root) {
 /////////////////////////// END ADVENTURE PACK CODE ////////////////////////////"
 `;
 
+exports[`App can equip single goody: JavaScript traverseLevelOrder 1`] = `
+"////////////////////////// BEGIN ADVENTURE PACK CODE ///////////////////////////
+// Adventure Pack commit fake-commit-hash
+// Running at: https://example.com/
+
+function* traverseLevelOrder(root) {
+  if (!root) {
+    return;
+  }
+
+  let level = [root];
+  while (level.length > 0) {
+    yield level;
+    level = level.flatMap((node) => [node.left, node.right].filter(Boolean));
+  }
```

Comment:
Since we know the first level will be non-empty at this point, we can switch to a do-while. Also, we can filter out the nulls after the map, I think that will create fewer intermediate arrays.

```suggestion
  do {
    yield level;
    level = level.flatMap((node) => [node.left, node.right]).filter(Boolean);
  } while (level.length > 0);
```

Reply:
ack!